### PR TITLE
MERGE WITH ATTENTION werf refactoring for using same parts with cse and oss

### PR DESCRIPTION
**Pull request by name212**
## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: <kebab-case of a module name> | <1st level dir in the repo>
type: fix | feature | chore
summary: <ONE-LINE of what effectively changes for a user>
impact: <what to expect for users, possibly MULTI-LINE>, required if impact_level is high ↓
impact_level: default | high | low
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by name212** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.29 12506196371 12506196371-1-con-1-29 /sys/deckhouse-oss/install:pr11275 redos@84.252.128.86

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by name212** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.29 12506631214 12506631214-1-con-1-29 /sys/deckhouse-oss/install:pr11275 redos@51.250.4.132

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by name212** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.29 12509113877 12509113877-1-con-1-29 /sys/deckhouse-oss/install:pr11275 redos@51.250.76.124

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/static'...

File: .werf/defines/base.tmpl [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  add: /tini/tini-static
+  to: /usr/bin/tini
+  before: setup
+- image: common/semver-artifact
+  add: /usr/local/bin/semver
+  to: /usr/local/bin/semver
+  before: setup
+  {{- range $k8sVersion := $context.k8sVersions }}
+    {{- $image_version := printf "%s.%d" $k8sVersion (index $context.CandiVersionMap "k8s" $k8sVersion "patch") | replace "." "-" }}
+- artifact: common/kubernetes-artifact-{{ $image_version }}
```
Comment (sprait) [inline .werf/defines/base.tmpl:18]:
```suggestion
- image: common/kubernetes-artifact-{{ $image_version }}
```

File: .werf/defines/base.tmpl [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  - lib64/libexpat.so*
+  - lib64/libssl.so.*
+  - lib64/libcrypto.so.*
+  - etc/pki
+  - usr/share/ca-certificates/ca-bundle.crt
+  - usr/bin/python3
+  - usr/bin/python3.12
+  - usr/lib/python3
+  - usr/lib64/python3
+  - usr/lib64/python3.12
```
Comment (sprait) [inline .werf/defines/base.tmpl:93]: maybe we can use statically compiled python here?

File: .werf/defines/base.tmpl [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+#   prefix - Files.Get prefix
+#   k8sVersions - list versions k8s to kubectl import
+#   CandiVersionMap - dict CandiVersionMap from werf.yaml
+#    - dict CandiVersionMap from werf.yaml
+{{- define "common_base_image" }}
+{{- $context := . -}}
+
+image: common-base
+fromImage: common/alt-p11
+import:
```
Comment (sprait) [inline .werf/defines/base.tmpl:77]:
```suggestion
image: common-base
fromImage: common/alt-p11
final: false
import:
```

File: .werf/defines/base.tmpl [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+#   rootContext - root context contains .Files
+#   prefix - Files.Get prefix
+#   k8sVersions - list versions k8s to kubectl import
+#   CandiVersionMap - dict CandiVersionMap from werf.yaml
+{{- define "base_for_go_image" }}
+{{- $context := . -}}
+
+image: base-for-go
+fromImage: common/alt-p11-artifact
+import:
```
Comment (sprait) [inline .werf/defines/base.tmpl:131]:
```suggestion
fromImage: common/alt-p11-artifact
final: false
import:
```

File: .werf/werf-release-channel.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
+---
+image: release-channel-version-prebuild
+from: {{ .Images.BASE_ALPINE_DEV }}
+dependencies:
```
Comment (sprait) [inline .werf/werf-release-channel.yaml:4]:
```suggestion
from: {{ .Images.BASE_ALPINE_DEV }}
final: false
dependencies:
```

File: .werf/werf-version-map.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
+---
+artifact: version-map-artifact
+from: {{ .Images.BASE_ALPINE }}
```
Comment (sprait) [inline .werf/werf-version-map.yaml:3]:
```suggestion
image: version-map-artifact
final: false
from: {{ .Images.BASE_ALPINE }}
```

File: .werf/werf-golang-ci-lint.yaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+---
+artifact: golangci-lint-artifact
+from: {{ $.Images.BASE_GOLANG_23_ALPINE }}
```
Comment (sprait) [inline .werf/werf-golang-ci-lint.yaml:3]:
```suggestion
image: golangci-lint-artifact
final: false
from: {{ $.Images.BASE_GOLANG_23_ALPINE }}
```

File: .werf/werf-golang-ci-lint.yaml [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+  beforeInstall:
+  - apk add --no-cache git openssh-client
+  - mkdir -p ~/.ssh && echo "StrictHostKeyChecking accept-new" > ~/.ssh/config
+  install:
+  # Use v1.60.1 for go1.23 - https://github.com/golangci/golangci-lint/issues/4837
+  - export GOPROXY={{ .GOPROXY }}
+  - git clone --depth 1 {{ .SOURCE_REPO }}/golangci/golangci-lint --branch v1.60.1
+  - cd golangci-lint/
+  - CGO_ENABLED=0 GOOS=linux go build -ldflags '-s -w -extldflags "-static"' -o /usr/local/bin/golangci-lint cmd/golangci-lint/main.go
+---
```
Comment (sprait) [inline .werf/werf-golang-ci-lint.yaml:14]:
```suggestion
  - git clone --depth 1 {{ .SOURCE_REPO }}/golangci/golangci-lint --branch v1.60.1
  - cd golangci-lint/
  - CGO_ENABLED=0 GOOS=linux go build -ldflags '-s -w -extldflags "-static"' -o /usr/local/bin/golangci-lint cmd/golangci-lint/main.go
  - rm -rf /golangci-lint/.git
---
```

File: .werf/defines/ssh-static.tmpl [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+{{- define "ssh_static_artifact" }}
+{{- $context := . -}}
+
+---
+image: ssh-static-src-artifact
+final: false
+fromImage: common/src-artifact
+shell:
+  install:
+  - git clone -b V_9_9_P1 --single-branch --depth=1 {{ $context.SOURCE_REPO }}/openssh/openssh-portable /src
```
Comment (sprait) [inline .werf/defines/ssh-static.tmpl:14]:
```suggestion
  install:
  - git clone -b V_9_9_P1 --single-branch --depth=1 {{ $context.SOURCE_REPO }}/openssh/openssh-portable /src
  - rm -rf /src/.git
```

File: .werf/defines/installer.tmpl [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+
+- artifact: terraform # from modules/040-terraform-manager/images/terraform-manager-base/werf.inc.yaml
+  add: /terraform/terraform
+  to: {{ $prefix }}/bin/terraform
+  before: setup
+{{- range $_, $edition := $context.Editions }}
+  {{- if $edition.terraformProviders }}
+    {{- range $_, $tfProvider := $edition.terraformProviders }}
+      {{- $tf := index $context.TF $tfProvider }}
+- artifact: {{ $tf.artifact }} # from modules/040-terraform-manager/images/terraform-manager-{PROVIDER}/werf.inc.yaml
```
Comment (sprait) [inline .werf/defines/installer.tmpl:80]:
```suggestion
- image: terraform # from modules/040-terraform-manager/images/terraform-manager-base/werf.inc.yaml
  add: /terraform/terraform
  to: {{ $prefix }}/bin/terraform
  before: setup
{{- range $_, $edition := $context.Editions }}
  {{- if $edition.terraformProviders }}
    {{- range $_, $tfProvider := $edition.terraformProviders }}
      {{- $tf := index $context.TF $tfProvider }}
- image: {{ $tf.artifact }} # from modules/040-terraform-manager/images/terraform-manager-{PROVIDER}/werf.inc.yaml
```

**Review by sprait — COMMENTED** [review]
minor suggestions

File: .werf/defines/base.tmpl (hunk 2 above)
Comment (name212) [reply .werf/defines/base.tmpl:93]: in the future

**Pull request merged**